File: crates/libs/lib-utils/src/validators.rs
pub fn validate_string(s: String) -> bool {
    let len = s.len();
    if len <= 3 || len >= 32 || s.contains(' ') {
        return false;
    }

    let allowed_characters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ\
                               абвгдеёжзиклмнопрстуфхцчшщъыьэюя-_@#$%!*0123456789";

    if s.chars().all(|c| c.is_digit(10)) {
        return false;
    }

    s.chars().all(|c| allowed_characters.contains(c))
}

File: bot/utils/validators.py
def validate_string(s: str) -> bool:
    if 3 < len(s) < 32 and " " not in s:
        allowed_characters = set(
            "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
            "абвгдеёжзиклмнопрстуфхцчшщъыьэюя"
            "-_@#$%!*0123456789"
        )

        if s.isdigit():
            return False

        return all(char in allowed_characters for char in s)

    return False
